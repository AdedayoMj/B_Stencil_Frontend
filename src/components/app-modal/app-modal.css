.modal {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 9999;
  opacity: 0;
  pointer-events: none;
  transition: opacity 0.3s ease;
  padding: 1rem;
}

.modal.show {
  opacity: 1;
  pointer-events: auto;
}

.modal-content {
  position: relative;
  max-width: 350px;
  width: 100%;
  background: var(--app-background);
  color: var(--app-text);
  padding: 20px;
  border-radius: 5px;
}

.close {
  position: absolute;
  top: 10px;
  right: 20px;
  font-size: 25px;
  cursor: pointer;
}
