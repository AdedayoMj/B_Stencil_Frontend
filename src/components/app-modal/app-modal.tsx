import { Component, Prop, State, h, Element, Watch, Listen } from '@stencil/core';

@Component({
  tag: 'app-modal',
  styleUrl: 'app-modal.css',
  shadow: true,
})
export class AppModal {
  @Prop() isOpen = false;

  @State() showModal = false;

  @Element() hostElement: HTMLElement;

  componentWillLoad() {
    this.showModal = this.isOpen;
  }

  @Watch('isOpen')
  updateOpenState(newValue: boolean) {
    this.showModal = newValue;
  }

  handleClose() {
    this.showModal = false;
    const closeEvent = new CustomEvent('modalClose', { bubbles: true, composed: true });
    this.hostElement.dispatchEvent(closeEvent);
  }

  @Listen('keydown')
  handleKeyDown(event: KeyboardEvent) {
    if (event.key === 'Escape' || event.key === 'Esc') {
      event.preventDefault();
      this.handleClose();
    }
  }

  render() {
    return (
      <div class={`modal ${this.showModal ? 'show' : ''}`} role="dialog" aria-modal={this.showModal.toString()} tabIndex={-1}>
        <div class="modal-content">
          <span class="close" onClick={() => this.handleClose()}>
            &times;
          </span>
          <slot></slot>
        </div>
      </div>
    );
  }
}
