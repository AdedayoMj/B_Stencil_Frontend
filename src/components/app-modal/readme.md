# app-modal

<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description | Type      | Default |
| -------- | --------- | ----------- | --------- | ------- |
| `isOpen` | `is-open` |             | `boolean` | `false` |


## Dependencies

### Used by

 - [expense-form](../expense-form)

### Graph
```mermaid
graph TD;
  expense-form --> app-modal
  style app-modal fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
