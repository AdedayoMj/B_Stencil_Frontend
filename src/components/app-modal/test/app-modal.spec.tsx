import { newSpecPage } from '@stencil/core/testing';
import { AppModal } from '../app-modal';

describe('AppModal', () => {
  it('should close the modal when close button is clicked', async () => {
    const page = await newSpecPage({
      components: [AppModal],
      html: `<app-modal></app-modal>`,
    });

    const appModal = page.rootInstance as AppModal;
    const closeButton = page.root.shadowRoot.querySelector('.close') as HTMLElement;

    // Initial state should be closed
    expect(appModal.showModal).toBe(false);
    expect(page.root).toMatchSnapshot();

    // State should be open
    appModal.showModal = true;
    await page.waitForChanges();

    // State should be open
    expect(appModal.showModal).toBe(true);
    expect(page.root).toMatchSnapshot();

    // Click the close buttonf
    closeButton.click();
    await page.waitForChanges();

    // State should be closed again
    expect(appModal.showModal).toBe(false);
    expect(page.root).toMatchSnapshot();
  });
});
