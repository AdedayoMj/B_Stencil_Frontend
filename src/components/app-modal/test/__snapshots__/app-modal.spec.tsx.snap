// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppModal should close the modal when close button is clicked 1`] = `
<app-modal>
  <mock:shadow-root>
    <div aria-modal="false" class="modal" role="dialog" tabindex="-1">
      <div class="modal-content">
        <span class="close">
          ×
        </span>
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
</app-modal>
`;

exports[`AppModal should close the modal when close button is clicked 2`] = `
<app-modal>
  <mock:shadow-root>
    <div aria-modal="true" class="modal show" role="dialog" tabindex="-1">
      <div class="modal-content">
        <span class="close">
          ×
        </span>
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
</app-modal>
`;

exports[`AppModal should close the modal when close button is clicked 3`] = `
<app-modal>
  <mock:shadow-root>
    <div aria-modal="false" class="modal" role="dialog" tabindex="-1">
      <div class="modal-content">
        <span class="close">
          ×
        </span>
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
</app-modal>
`;
